src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -229,6 +230,11 @@ internal CookieContainer CookieContainer
             set
             {
                 CheckDisposedOrStarted();
+                if (value == null)
+                {
+                    throw new ArgumentNullException("CookieContainer", SR.net_http_invalid_cookiecontainer);
```

Comment:
I understand.  Desktop does throw an exception.  For now, you should match HttpClientHandler/WinHttpHandler behavior for CoreFx which is to NOT throw an exception in the setter.

But I think this is an interesting app-compat issue to think about more.  Please open up an issue and assign it to me so that we can track getting to the right behavior.  We may end up putting back the exception in the setter but I would like to do it consistently across all platforms for CoreFx.